### PR TITLE
feat(timm_resnet): register the resnet18 and resnext50 checkpoints

### PR DESCRIPTION
**Pull request by zhenshanx-nv**
> **Stacked on #1121.** That pull request adds the `timm_resnet` family; this
> one only registers two further checkpoints. The diff below includes #1121
> until it merges, after which this branch will be rebased.

## Background

#1121 adds the `timm_resnet` family with `timm/resnet50.a1_in1k` as its only
registered checkpoint. The builder recovers the stage layout from the
checkpoint rather than from a depth table, so the basic-block and
grouped-convolution variants already build correctly and were verified during
that work, but neither is registered as a supported model.

## Exit Criteria

- `timm/resnet18.a1_in1k` and `timm/resnext50_32x4d.a1h_in1k` are registered
  checkpoints with E2E manifests, validation workloads, and performance
  coverage.
- No builder change is required.

Non-goal: `resnet34`, `resnet101`, `resnet152`, and `wide_resnet` also match the
family prefixes but are not registered here.

## Implementation

Registration only. Adds E2E manifests and thresholds for both checkpoints,
registers them in the support matrix, HF metadata, and validation workloads,
and adds them to the release performance suite as `additional_profiles`
inheriting `timm_resnet.classify`. Repository counters are updated to match.

No source change to the family: `python/tensorrt_model_connect/families/timm_resnet`
and `src/runtime/models/timm_resnet` are untouched by this pull request.

### Change categories

- [x] Model or runtime behavior
- [ ] Public API
- [ ] ABI
- [ ] Bundle or artifact format
- [ ] Dependencies
- [ ] Documentation only
- [ ] CI or developer tooling

### Commands and Results

```
python -m pytest tests/builder/ tests/tools/ tests/e2e_harness/ -q -n 8 \
  --dist=worksteal --import-mode=importlib -p no:cacheprovider
=> 3948 passed, 8 skipped

python -m ruff check tests/e2e/models/timm_resnet   => All checks passed
python tools/legal_headers.py                       => findings=0
```

Numerical parity, measured when the family landed. Engine logits compared
against an independent PyTorch reference that applies batch norm unfolded:

| Checkpoint | Layout discovered | Correlation | argmax | top-5 |
| --- | --- | --- | --- | --- |
| `timm/resnet18.a1_in1k` | `[2,2,2,2]` basic | 0.99999826 | match | 5/5 |
| `timm/resnext50_32x4d.a1h_in1k` | `[3,4,6,3]` grouped | 0.99999678 | match | 5/5 |

### Hardware, Environment, and Revisions

- GPU: NVIDIA A100-SXM4-80GB, compute capability 8.0.
- Container: `Dockerfile.dev.x86` dev image, Ubuntu 24.04, Python 3.12.
- TensorRT 11.1.0.106, CUDA architecture `80-real`, Release build.
- Parity measured at fp32; both checkpoints are registered for `FP16` builds.
- Checkpoint revisions at time of testing:
  - `timm/resnet18.a1_in1k` @ `491b427b45c94c7fb0e78b5474cc919aff584bbf`
  - `timm/resnext50_32x4d.a1h_in1k` @ `cc736cb0940158d05966023116d90fca4c38dae3`

  The manifests intentionally do not pin `hf_revision`: the timm reference
  resolves `hf-hub:<id>` at `main`, so a pin would disagree with the cache the
  warm step populates and fail the offline reference run. See #1121.

### Not Run / Remaining Gaps

- No E2E harness run for either new checkpoint. Both are registered and will be
  exercised by the model proof; the parity evidence above comes from a direct
  engine-versus-PyTorch comparison rather than the harness.
- No performance numbers. The profiles are registered but were not run.
- `resnet34`, `resnet101`, `resnet152`, and `wide_resnet` remain unregistered
  and unverified.

## Notes For Future Readers

Adding a further ResNet depth should need only a manifest, the shared
registration entries, and a counter bump. The builder is layout-driven, so no
graph code changes.

Note that `tools/model_ci.py` reads manifests from the committed tree via
`git ls-tree`, so a new manifest must be committed before
`test_nightly_inventory_exactly_matches_every_model_proof_selection` will agree
with filesystem-based test selection.

### Risk level

- [x] Low
- [ ] Medium
- [ ] High

Registration only, no source change to the family, and the full CPU suite
passes.

### CODE REVIEW

**Comment by zhenshanx-nv** [comment]
Closing as obsolete: this registered checkpoints through the shared matrices and workload files, which #1188 deleted in favour of a per-family support.py. Re-registering resnet18 and resnext50 under the new layout is a separate, much smaller change against families/timm_resnet/.
